Mathlib/MeasureTheory/MeasurableSpace/Basic.lean

```diff
@@ -1182,8 +1183,67 @@ namespace MeasurableSpace
   rintro _ ⟨u, -, rfl⟩
   exact (show MeasurableSet s from GenerateMeasurable.basic _ <| mem_singleton s).mem trivial
 
+lemma generateFrom_singleton_le {m : MeasurableSpace α} {s : Set α} (hs : MeasurableSet s) :
+    MeasurableSpace.generateFrom {s} ≤ m :=
+  generateFrom_le (fun _ ht ↦ mem_singleton_iff.1 ht ▸ hs)
+
 end MeasurableSpace
 
+namespace MeasureTheory
+
+theorem measurableSet_generateFrom_singleton_iff {s t : Set α} :
+    MeasurableSet[MeasurableSpace.generateFrom {s}] t ↔ t = ∅ ∨ t = s ∨ t = sᶜ ∨ t = univ := by
```

Comment:
There is a shorter proof of that in the (closed) PR #16293
Also @YaelDillies wrote over there that we actually don't want that lemma. I'll let them explain why we wouldn't even though several people seem to need it and it seems very natural.

Reply:
I wasn't able to get the old proof to work so that's why rewrote the proof. Feel free to push a shorter proof if anyone can golf it.

I talked with @YaelDillies already and I think what they meant in that comment is that it was not needed for Violeta and not that it's not useful in general.